common/Session.cpp

```diff
@@ -318,11 +319,13 @@ void Session::getIOStats(uint64_t &sent, uint64_t &recv)
 
 void Session::dumpState(std::ostream& os)
 {
-    os << "\n\t\tid: " << _id
+    os << std::boolalpha
```

Comment:
When we have these state changers, should we restore them before leaving scope, or have them set at the stream construction site? There's already plenty of them, so not a new issue though.

Reply:
Good question. I've struggled with it myself for the longest time. The approach that I've assumed is to set what we need and assume any arbitrary state. This is because it's really hard to know/remember default states, and it's customary to set states per-preference. Having said that, I think we reset `hex` to `dec`, since it's obvious that we want `dec` by default and `hex` is the exception.

Not ideal, but since we only have a few of those (`boolalpha` and `hex` are by far the most common), it seems to be fine.

But we can change that. If you feel it's cleaner to always reset to a certain "default" (even if by agreement), it might be worth it.